client/components/themes-list/index.jsx

```diff
@@ -20,15 +23,39 @@ import { getSelectedSite } from 'calypso/state/ui/selectors';
 import './style.scss';
 
 const noop = () => {};
+const THEME_CARD_WIDTH = 320; // used to calculate the number of column in theme-list
+const THEM_CARD_MARGIN = 16; // used to calculate the max-width of column in theme-list
 
 export const ThemesList = ( props ) => {
+	const [ ctaPosition, setCtaPosition ] = useState( 0 );
+
 	const fetchNextPage = useCallback(
 		( options ) => {
 			props.fetchNextPage( options );
 		},
 		[ props.fetchNextPage ]
 	);
 
+	const handleContentRectChange = useCallback(
+		( contentRect, elementRef ) => {
+			if ( contentRect ) {
+				const rowCount = Math.floor( contentRect.width / THEME_CARD_WIDTH );
+				const relativeCtaPosition =
+					rowCount * 3 - 1 < props.themes.length - 1 ? rowCount * 3 - 1 : props.themes.length - 1;
+				setCtaPosition( relativeCtaPosition );
+				// Set the min-width varible for theme card
+				elementRef.current.style.setProperty(
+					'--theme-list-item-width',
```

Comment:
I noticed `.theme` [flex-grow:1](https://github.com/Automattic/wp-calypso/blob/0ab30f791331895435ee69707964511d76b5135c/client/components/theme/style.scss#L8) can lead to this case, so I decided to calculate the theme card width programmatically. Suggestions are welcome
| flex-grow: 1 | flex-grow: 0 | flex-grow: 0 with CSS variable |
|-------------------------|----------------------|----------------------------|
|              ![0t29VX.png](https://user-images.githubusercontent.com/10071857/215336512-31a52c16-7e8e-4fbf-b9be-2415e6f3d5aa.png)       |           ![DHcHd1.png](https://user-images.githubusercontent.com/10071857/215336634-c1df8cbb-f6b0-4d7d-9974-97099fce0bcc.png)        |           ![Zo7WDA.png](https://user-images.githubusercontent.com/10071857/215336665-c88668cd-7c13-47e5-a5b5-53de85fe117a.png)           |

Reply:
We decided to take advantage of <TrailingItems/> component and leave the flex-grow value as-is. More detail on p1675070281138659-slack-CRWCHQGUB